java/src/main/native/src/ColumnVectorJni.cpp

```diff
@@ -1262,6 +1281,40 @@ JNIEXPORT jlongArray JNICALL Java_ai_rapids_cudf_ColumnVector_getNativeDataPoint
   CATCH_STD(env, 0);
 }
 
+JNIEXPORT jint JNICALL Java_ai_rapids_cudf_ColumnVector_getNativeNumChildren(JNIEnv *env,
+                                                                           jobject j_object,
+                                                                           jlong handle) {
+
+    JNI_NULL_CHECK(env, handle, "native handle is null", 0);
+    try {
+      cudf::jni::auto_set_device(env);
+      cudf::column_view *column = reinterpret_cast<cudf::column_view *>(handle);
+      // Strings has children(offsets and chars) but not a nested child() we care about here.
+      if (column->type().id() == cudf::type_id::STRING) {
+        return 0;
+      }
+      // first child is always offsets which we do not want to count here
```

Comment:
@razajafri making doubly sure we cover strings although the java side wont let u come here per current change/design. Please review.